src/expressions/functions/builtInFunctions.string.ts

```diff
@@ -405,6 +405,32 @@ const fnNormalizeSpace: FunctionDefinitionType = function(
 	return sequenceFactory.singleton(createAtomicValue(string.replace(/\s+/g, ' '), 'xs:string'));
 };
 
+const fnTranslate: FunctionDefinitionType = (
+	_dynamicContext,
+	_executionParameters,
+	_staticContext,
+	arg1,
+	arg2,
+	arg3
+) => {
+	const arr1 = Array.from(arg1.first() ? arg1.first().value : '');
```

Comment:
Can we use `zipSingleton` for these three variables? This will make the code more future-proof for synchronous evaluation